### DISCUSSION TITLE
DatePickerInput's aria-label isn't prioritized by Playwright

### DISCUSSION THREAD

**Discussion by vorant94** [mantinedev/mantine#6412] [Q&A]
Hi, there.

First of all thanks for the great library! It is really a breath of the fresh air for me!

I have a problem while trying to use `DatePickerInput` inside my e2e tests with Playwright.  I use it like that

```tsx
<DatePickerInput
  label="Started At"
  placeholder="placeholder"
  value={value}
  onChange={onChange}
  onBlur={onBlur}
/>
```

Despite me passing `label` to the component and despite component is rendering the right `aria-label` Playwright still can find the trigger button with `getByLabel` only by its **placeholder text** or by **selected value** (if any) instead of **label** I set i.e.:

```tsx
page.getByLabel('4 June 2024', { exact: true }); // => true if there is a value
page.getByLabel('placeholder'); // => true if no value
page.getByLabel('started at'); // => always false
```

Maybe you have some idea on why is it so and how can I have some sort of static identifier for the date picker component that is not changing during user interactions?

Since I'm new to accessibility I'm not sure whether it is something that can be done on component lib side or it is just a bug on Playwright side... Sorry if its not related to you, therefore I opened a discussion and not issue

**Comment by vorant94** [comment]
I tried to reproduce the same stuff on a blank vite project and discovered that I initially didn't understand the issue enough

the reason for the behavior I described earlier is because `label` prop of `DatePickerInput` doesn't set `aria-label` on a button.

button's `aria-label` is equal to placeholder or value of picker unless explicit `aria-label` prop is passed like this 

```
<DatePickerInput
  label="Started At"
  aria-label="Started At"
  placeholder="Started At"
  value={value}
  onChange={onChange}
  onBlur={onBlur}
/>
```

so it isn't related to playwright and the docs of DatePickerInput (https://mantine.dev/dates/date-picker-input/#accessibility) isn't very clear since it says that passing `label` makes `aria-label` to be redundant in term of accessibility, but it appears that it isn't redundant

**Comment by starxyn** [comment]
Hi, 
Can I check if you get the Playwright to work using aria-label? Were you using `page.getByLabel('started at');` in playwright? Coz it doesn't work for me.

**Comment by vorant94** [comment]
@starxyn i haven't looked into this code for a relatively long time, but i do remember it worked for me after this [fix](https://github.com/mantinedev/mantine/pull/6530), so here are the files

https://github.com/vorant94/the-repo/blob/master/apps/subs-savvy/e2e/coms/date-picker-input.com.ts
https://github.com/vorant94/the-repo/blob/master/apps/subs-savvy/e2e/coms/upsert-subscription.com.ts
